app/version.go

```diff
@@ -109,3 +125,51 @@ func NewBuildInfo() BuildInfo {
 
 	return info
 }
+
+func checkForUpdate(buildInfo *BuildInfo) error {
+	latestInfo, err := fetchLatestVersion()
+	if err != nil {
+		return fmt.Errorf("failed to fetch latest version: %w", err)
+	}
+
+	if strings.Contains(buildInfo.Version, "no-version-available") {
+		fmt.Println("no-version-available detected, unable to determine if current build is latest version")
+	} else if buildInfo.Version == latestInfo.TagName {
+		fmt.Println("You are running the latest version")
+	} else {
+		fmt.Printf("A newer version is available: %s\n", latestInfo.TagName)
+	}
+
+	return nil
+}
+
+func fetchLatestVersion() (Release, error) {
+	url := "https://api.github.com/repos/temporalio/tcld/releases/latest"
+	resp, err := http.Get(url)
+	if err != nil {
+		return Release{}, err
+	}
+
+	defer func(Body io.ReadCloser) {
```

Comment:
nit: Body -> body (golang uses snakeCase naming convention)

Reply:
Went ahead and simplified the defer logic.